app/src/main/java/de/danoeh/antennapod/ui/screen/queue/QueueFragment.java

```diff
@@ -596,6 +613,66 @@ protected void onSelectionChanged() {
         }
     }
 
+    /**
+     * This method checks if the selected items are allowed to be moved to the top, bottom or both of the Queue.
+     * @param queue The FeedItems currently in the Queue.
+     * @param selectedItems The FeedItems for which the check is performed.
+     * @return A boolean array where:
+     *           [0] is true if moving to the top is allowed, false otherwise.
+     *           [1] is true if moving to the bottom is allowed, false otherwise.
+     * */
+    public static boolean[] canMove(List<FeedItem> queue, List<FeedItem> selectedItems) {
+        boolean canMoveToTop = true;
+        boolean canMoveToBottom = true;
+
+        if (selectedItems.isEmpty() || queue.isEmpty()) {
+            return new boolean[]{false, false};
+        }
+
+        // No manual reordering allowed if sorting enforced, or queue locked. Both move options are disabled.
+        if (UserPreferences.isQueueLocked() || UserPreferences.isQueueKeepSorted()) {
+            return new boolean[]{false, false};
+        }
+
+        int queueSize = queue.size();
+        int selectedSize = selectedItems.size();
+
+        boolean isFirstItemSelected = selectedItems.get(0).getId() == queue.get(0).getId();
+        boolean isLastItemSelected = selectedItems.get(selectedSize - 1).getId() == queue.get(queueSize - 1).getId();
+
+        // If all items in the list are selected, disable move options since no movement is possible or allowed.
+        if (selectedSize == queueSize) {
+            return new boolean[]{false, false};
+        }
+
+        if (selectedItems.size() == 1) {
+            // If the selected item is already at the top of the list, disable the option to move it to the top.
+            if (isFirstItemSelected) {
+                canMoveToTop = false;
+            }
+            // If the selected item is already at the bottom of the list, disable the option to move it to the bottom.
+            if (isLastItemSelected) {
+                canMoveToBottom = false;
+            }
```

Comment:
I think the code would be easier to understand if you directly add a return statement here instead of setting the variables. Something like `return new boolean[]{!isFirstItemSelected, !isLastItemSelected}`